config/FindCOMPTON.cmake

```diff
@@ -6,86 +6,192 @@
 # note   Copyright (C) 2016-2017 Los Alamos National Security, LLC.
 #        All rights reserved.
 #------------------------------------------------------------------------------#
-# - Find Compton
-# Find the native Compton includes and library
+
+#.rst:
+# FindCOMPTON
+# ---------
+#
+# Find the CSK COMPTON includes and libraries.
+#
+# CSK COMPTON is a project dedicated to the calculation and interpolation of
+# Compton Scattering Kernel (CSK, hence the project name) values for use in
+# radiative transfer simulations.  https://gitlab.lanl.gov/keadyk/CSK_generator
+#
+# Imported Targets
+# ^^^^^^^^^^^^^^^^
+#
+# If COMPTON is found, this module defines the following :prop_tgt:`IMPORTED`
+# targets::
+#
+#  COMPTON::compton         - The COMPTON library.
+#
+# Result Variables
+# ^^^^^^^^^^^^^^^^
+#
+# This module will set the following variables in your project::
+#
+#  COMPTON_FOUND          - True if COMPTON found on the local system
+#  COMPTON_INCLUDE_DIRS   - Location of COMPTON header files.
+#  COMPTON_LIBRARIES      - The COMPTON libraries.
+#  COMPTON_VERSION        - The version of the discovered COMPTON install.
+#
+# Hints
+# ^^^^^
 #
-#  COMPTON_INCLUDE_DIRS   - where to find multigroup_lib_builder.hh, etc.
-#  COMPTON_LIBRARIES      - List of libraries when using Compton.
-#  COMPTON_FOUND          - True if Compton found.
+# Set ``COMPTON_ROOT_DIR`` to a directory that contains a COMPTON installation.
+#
+# This script expects to find libraries at ``$COMPTON_ROOT_DIR/lib`` and the
+# COMPTON headers at ``$COMPTON_ROOT_DIR/include``.  The library directory may
+# optionally provide Release and Debug folders.
+#
+# Cache Variables
+# ^^^^^^^^^^^^^^^
+#
+# This module may set the following variables depending on platform and type of
+# COMPTON installation discovered.  These variables may optionally be set to
+# help this module find the correct files::
+#
+#  COMPTON_LIBRARY        - Location of the COMPTON library.
+#  COMPTON_LIBRARY_DEBUG  - Location of the debug COMPTON library (if any).
+#
+#------------------------------------------------------------------------------#
+
+# Include these modules to handle the QUIETLY and REQUIRED arguments.
+include(FindPackageHandleStandardArgs)
 
-if (APPLE)
-    set(CMAKE_FIND_LIBRARY_SUFFIXES ".a;.dylib")
+#=============================================================================
+# If the user has provided ``COMPTON_ROOT_DIR``, use it!  Choose items found
+# at this location over system locations.
+if( EXISTS "$ENV{COMPTON_ROOT_DIR}" )
```

Comment:
This variable should be set by the module file, right? Is that what you mean by "if the user has provided COMPTON_ROOT_DIR"? (I just want to be sure I understand what's going on.)

Reply:
The module sets this variable (or it could be set w/o a module).  Ideally, the module should set `CMAKE_PREFIX_PATH` instead.  A CMake variable of this same name will also work.